src/backend/bacon-video-widget-gst-missing-plugins.c

```diff
@@ -521,30 +589,31 @@ show_codec_missing_information_dialog (TotemCodecInstallContext *ctx)
 	                                 GTK_MESSAGE_ERROR,
 	                                 GTK_BUTTONS_CANCEL,
 	                                 _("Unable to play the file"));
+	action = determine_vlc_launch_action_for_missing_codecs (bacon_video_widget_get_mrl (ctx->bvw));
 
-	/* Show additional button pointing to our online store if the
-	 * missing codec can be installed via our codecs activation key. */
-	if (missing_codecs_supported ((const char**)ctx->details)) {
-		GtkWidget *button;
-		const char *button_text;
-
-		message_text = _("This file type is currently unsupported. To support this type "
-				 "of file, you can purchase our audio/video codec upgrade to add "
-				 "support for additional file formats.");
-
-		button_text = _("_Purchase Codec Upgrade");
-		button = gtk_dialog_add_button (GTK_DIALOG (dialog),
-						button_text,
-						GTK_RESPONSE_ACCEPT);
+	if (action) {
+		GtkWidget *button = gtk_dialog_add_button (GTK_DIALOG (dialog),
+							   action->button_text,
+							   GTK_RESPONSE_ACCEPT);
+		message_text = action->message_text;
 		gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_ACCEPT);
 
 		gtk_style_context_add_class (gtk_widget_get_style_context (button), "suggested-action");
+		g_signal_connect_data (dialog,
+				       "response",
+				       G_CALLBACK (codec_confirmation_dialog_response_cb),
+				       action,
+				       (GClosureNotify) codec_missing_suggested_launch_action_free,
+				       G_CONNECT_AFTER);
 	} else {
-		message_text = _("This file type is unsupported.");
+		message_text = _("There are no applications that can play this file");
```

Comment:
Not sure what to do in this case - the user would have had to somehow remove `eos-vlc.desktop` from their applications directory. Probably showing the codec upgrade here is wrong, since that never worked anyway.